scripts/ampHtmlValidator/index.js

```diff
@@ -1,5 +1,4 @@
 /* eslint-disable no-console */
-const fetch = require('isomorphic-fetch');
```

Comment:
why don't we need to [import `fetch` from `undici`](https://github.com/nodejs/undici#undicifetchinput-init-promise) here?

Reply:
'fetch' is available globally in Node, so it doesn't need imported from v18 onwards.

Reply:
Node doesn't expose `setGlobalDispatcher` or `Agent` globally, so we need to install `undici` as a dependency to be able to use these for our certs.

Reply:
> Node doesn't expose `setGlobalDispatcher` or `Agent` globally, so we need to install `undici` as a dependency to be able to use these for our certs.

hmm that's a bit odd, could we not be using components from one version of `undici` that are incompatible with the version baked into Node? Seems like node should expose the entire dependency to me, this does feel a bit like an unstable API with these sorts of inconsistencies..

In practice I would expect any major problems, just feels a bit messy. Is there any argument to say we should just use `undici` and turn off experimental fetch; that would mean we use `undici` as per the docs for now?

Reply:
I think I read on one of their Git issues that they were considering exposing more through Node. I'll see if I can find that.

We could probably use `undici` fetch from the dependency, although that would likely bloat the bundle size again since we'd need to import `fetch` from a dependency rather than Node core.

Reply:
Couple of bits I came across:
`https://github.com/orgs/nodejs/discussions/44038#discussioncomment-5701073`

`https://github.com/nodejs/node/issues/43187`

`https://stackoverflow.com/questions/77381314/is-it-possible-to-set-a-global-dispatcher-for-the-built-in-fetch-in-node`

Reply:
I would hope that it would follow most other dependencies whereby its stated as a **breaking change** when the minimum version of Node changes when its updated.

Reply:
> I would hope that it would follow most other dependencies whereby its stated as a **breaking change** when the minimum version of Node changes when its updated.

Oh I didn't think about it that way, I suppose if `undici` make incompatible change with the fetch baked into node they _should_ up move to a newer version of Node that it is compatible with.

> We could probably use undici fetch from the dependency, although that would likely bloat the bundle size again since we'd need to import fetch from a dependency rather than Node core.

I wouldn't expect it to have any effect on the clientside bundle as fetch should not rely on `undici` at all on the client?

I think if using `undici` from a dependency rather than node has no negative performance impacts I feel it may be less confusing to not use the experimental API as we don't have to explain:
- Node has a hidden dependency on `undici`
- We have to import `undici` to fill in gaps in the baked in node version `undici`

Be interesting to hear others thoughts?

Reply:
The client bundling is a big weird/broken I think. Server-only code does make its way into client bundle I've noticed. The `fetchPageData.js` being one of those functions.

If I `import { fetch } from 'undici';` in that file, it requires me to update the `webpack.config.client.js` file to `resolve` fallbacks for server only modules, like `tls`, `node:events` as reverted in: https://github.com/bbc/simorgh/pull/11122/commits/1b3ddb91f565869d657939fc3351e018769d4b7a

The bundle size jumps up as well from:
```
MIN_SIZE = 771   -> 1140
MAX_SIZE = 1228  -> 1597
```

The behaviour is strange. Those `getInitialData` functions are imported at the root of each page type, so _possibly_ they are considered client modules?

Reply:
> Oh I didn't think about it that way, I suppose if undici make incompatible change with the fetch baked into node they should up move to a newer version of Node that it is compatible with.

Yea thats my thinking. I would like to think they would do that, but who knows 😅

Reply:
> The client bundling is a bit weird/broken I think. Server-only code does make its way into client bundle I've noticed. The `fetchPageData.js` being one of those functions.

hmm that sounds pretty weird and broken and definitely worth investigating outside of this 🤔 We may well have made assumptions when we planned to do clientside navigation that no longer hold true now.

Given the bundlesize jump for this PR I'd definitely stick with experimental api approach as we stay secure and up-to-date but take a bit of a hit on codebase simplicity 😞 

Should we get tech ticket together to investigate this bundling weirdness, I feel we will still be using our Express for a little while yet so we should definitely invest some time in this? You removing react-router and the scaffolding for clientside nav means we are more likely to have some positive results in this area now.

Reply:
Yea definitely worth looking into. Its a good point around the history of client-side routing, could very well have something to do with the setup.

I'll get it ticketed up 👍

Reply:
Thank you!